### ISSUE TITLE
Kingsmith G1 Walking Pad

### ISSUE THREAD

**Issue by cagnulein** [cagnulein/qdomyos-zwift#1833]
![IMG_7316](https://github.com/cagnulein/qdomyos-zwift/assets/48216/3c47bbb9-6520-487b-8a11-67238e9da4ec)

**Comment by cagnulein** [comment]
![D03192B4-5030-4065-B890-19FA6B0B09BC](https://github.com/cagnulein/qdomyos-zwift/assets/48216/26351171-3531-42c5-ac71-03cca47471ac)

**Comment by cagnulein** [comment]
[debug-Mi__Nov__29_07_40_36_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13498427/debug-Mi__Nov__29_07_40_36_2023.log)
[debug-Mi__Nov__29_07_39_46_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13498429/debug-Mi__Nov__29_07_39_46_2023.log)
[debug-Mi__Nov__29_07_38_54_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13498430/debug-Mi__Nov__29_07_38_54_2023.log)
[debug-Mi__Nov__29_07_37_34_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13498431/debug-Mi__Nov__29_07_37_34_2023.log)

@d3m3vilurr new encryption table , i'm trying to map it with chatgpt if i can

**Comment by d3m3vilurr** [comment]
sorry. I just read it now.
I'll check it tonight.

**Comment by cagnulein** [comment]
No worries! Thanks!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 15 dic 2023 alle 07:06 Sunguk Lee ***@***.***>
ha scritto:

**Comment by d3m3vilurr** [comment]
tl;dr.
probably key is `ZaCw4FGHIJqLhN+P8RVTU/WcY6ObDdefgEijklmnopQrsBuvMxXz1yA2t5079KS3=`
swapped 8 <-> 9 from original table (`ZaCw4FGHIJqLhN+P9RVTU/WcY6ObDdefgEijklmnopQrsBuvMxXz1yA2t5078KS3=`)

---- 

long answer

follow the steps:
1. in the log `debug-Mi__Nov__29_07_40_36_2023.log`
> << 41 36 63 36 6c 62 6e 38 67 52 63 4a 58 62 32 49 67 49 6b 2f 58 44 6d 4b 58 38 41 4b 6b 36 56 49 67 4c 54 55 4d 68 77 34 3d 0d
> << decrypted: even\u007F Error \"Error�ode\" -5001

i think this string could be `event Error "ErrorCode" -5001`

2. generate encrypted string with original table
> \> e64 event Error "ErrorCode" -5001
> b'6c6lbn9gRcJXb2IgIk/XDmKX9AKk6VIgLTUMhw4='

3. unhashed received buf(to base64 string)
> \> uh 36 63 36 6c 62 6e 38 67 52 63 4a 58 62 32 49 67 49 6b 2f 58 44 6d 4b 58 38 41 4b 6b 36 56 49 67 4c 54 55 4d 68 77 34 3d
> b'6c6lbn8gRcJXb2IgIk/XDmKX8AKk6VIgLTUMhw4='

4. compare these two string
> 6c6lbn9gRcJXb2IgIk/XDmKX9AKk6VIgLTUMhw4=
> 6c6lbn**8**gRcJXb2IgIk/XDmKX**8**AKk6VIgLTUMhw4=

5. so we should know what char is swapped in the table.

6. recheck some bufs of the logs which is not decrypted with new found key.
> \> k ZaCw4FGHIJqLhN+P8RVTU/WcY6ObDdefgEijklmnopQrsBuvMxXz1yA2t5079KS3= 
> \> d64 6c6lbn8gRcJXb2IgIk/XDmKX8AKk6VIgLTUMhw4=
> b'event Error "ErrorCode" -5001'
> \> d 64 6d 2f 58 44 41 6c 76 62 69 5a 4d 68 77 5a 31 77 38 3d 3d 0d
> b'version 0004\r'
> \> d 44 41 2f 58 64 6d 2f 58 44 58 5a 4d 77 38 3d 3d 0d
> b'servers 0\r'

**Comment by cagnulein** [comment]
Wow, thank you so much as always!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 17 dic 2023 alle 04:03 Sunguk Lee ***@***.***>
ha scritto:
